server/bunch/views.py

```diff
@@ -0,0 +1,144 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.shortcuts import get_object_or_404
+from rest_framework import permissions, status, viewsets
+from rest_framework.decorators import action
+from rest_framework.exceptions import *
+from rest_framework.response import Response
+
+from .models import Bunch, Channel, Member
+from .permissions import IsBunchMember, IsBunchOwner  # IsBunchAdmin
+from .serializers import BunchSerializer, ChannelSerializer, MemberSerializer
+
+User = get_user_model()
+
+
+class BunchViewSet(viewsets.ModelViewSet):
+    serializer_class = BunchSerializer
+    permission_classes = [permissions.IsAuthenticated, IsBunchOwner]
+    lookup_field = 'id'
+
+    def get_queryset(self):
+        all_bunches = Bunch.objects.all()
+        user_memberships = self.request.user.bunch_memberships.all()
+        queryset = Bunch.objects.filter(members__user=self.request.user).order_by('-created_at')
+
+        if not queryset.exists() and all_bunches.exists():
+            if not user_memberships.exists():
+                # for testing -- return all bunches if the user not part of any
+                return all_bunches.order_by('-created_at')
+        
+        return queryset
+
```

Comment:
Add dynamic permission for different endpoints, for example...

```suggestion
    @override
    def get_permissions(self):
        if self.action == "join":
            self.permission_classes = [permissions.IsAuthenticated]
        else:
            self.permission_classes = [permissions.IsAuthenticated, IsBunchOwner]

        return super().get_permissions()
```

Make suitable changes for all ViewSets